contrib/opentimelineio_contrib/adapters/otio_svg.py

```diff
@@ -0,0 +1,481 @@
+        separator = '\n'
+        image = header + '\n' + background + '\n' + font + '\n' + separator.join(
+            self.lines) + '\n</svg>'
+        svg_file = open(file_path, 'w')
+        svg_file.write(image)
+        svg_file.close()
+
+
+class ClipData:
+    src_start = 0.0
+    src_end = 0.0
+    avlbl_start = 0.0
+    avlbl_end = 0.0
+    avlbl_duration = 0.0
+    trim_start = 0.0
+    trim_duration = 0.0
+    target_url = ''
+    clip_id = 0
+
+    def __init__(self, src_start=0.0, src_end=0.0, avlbl_start=0.0,
+                 avlbl_end=0.0, avlbl_duration=0.0,
+                 trim_start=0.0, trim_duration=0.0, target_url='', clip_id=0):
+        self.src_start = src_start
+        self.src_end = src_end
+        self.avlbl_start = avlbl_start
+        self.avlbl_end = avlbl_end
+        self.avlbl_duration = avlbl_duration
+        self.trim_start = trim_start
+        self.trim_duration = trim_duration
+        self.target_url = target_url
+        self.clip_id = clip_id
+
+
+def convert_otio_to_svg(timeline, filepath):
+    renderer = SVGRenderer(2406.0, 1054.0)
+    image_margin = 10.0
+    font_size = 15
+    total_duration = 0
+    min_time = 0.0
+    max_time = 0.0
+    all_clips_data = []
+    clip_count = -1
+    for curr_clip in timeline.tracks[0]:
+        if isinstance(curr_clip, otio.schema.Clip):
+            avlbl_start = total_duration - curr_clip.source_range.start_time.value
+            min_time = min(min_time, avlbl_start)
+            src_start = total_duration
+            total_duration += curr_clip.source_range.duration.value
+            src_end = total_duration - 1
+            avlbl_end = (curr_clip.media_reference.available_range.start_time.value +
+                         curr_clip.media_reference.available_range.duration.value -
+                         curr_clip.source_range.start_time.value -
+                         curr_clip.source_range.duration.value + total_duration - 1)
+            max_time = max(max_time, avlbl_end)
+            trim_start = curr_clip.source_range.start_time.value
+            trim_duration = curr_clip.source_range.duration.value
+            avlbl_duration = curr_clip.media_reference.available_range.duration.value
+            clip_count += 1
+            clip_data = ClipData(src_start, src_end, avlbl_start,
+                                 avlbl_end, avlbl_duration, trim_start,
+                                 trim_duration,
+                                 curr_clip.media_reference.target_url, clip_count)
+            all_clips_data.append(clip_data)
+
+    scale_x = (renderer.width - (2.0 * image_margin)) / (max_time - min_time + 1.0)
+    x_origin = (-min_time) * scale_x
+
+    clip_rect_ht = (renderer.height - (2.0 * image_margin) - (2.0 * font_size)) / \
+                   ((4.0 + len(all_clips_data)) * 2 - 1)
+
+    # Draw Timeline
+    timeline_origin = Point(x_origin, renderer.height - image_margin - clip_rect_ht)
```

Comment:
Looking at the pattern you've used for the body of this function, this looks like a series of blocks that correspond to specific types in OTIO.  You might benefit from a dispatch table style pattern, similar to what is implemented in the rv adapter here: https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/afc69b96c7908b29c2334994e6c4404073d95fc7/contrib/opentimelineio_contrib/adapters/extern_rv.py#L70

The major benefit is that you become agnostic to the structure of the otio file.  It also becomes much easier to exend over time and handle things (like multiple tracks).